### PR TITLE
`Editable project location` and `Required-by` for `pip show`

### PR DESCRIPTION
**Pull request by eth3lbert**
## Summary

- Displays missing packages as single-line warnings.
- Adds support for `Editable project location` and `Required-by` fields in `pip show`.

Part of #2526.

### CODE REVIEW

File: crates/uv/src/commands/pip_show.rs [hunk 1]
```diff
@@ -120,21 +157,36 @@ pub(crate) fn pip_show(
                 .simplified_display()
         )?;
 
+        if let Some(url) = distribution.as_editable() {
+            let path = url.to_file_path().unwrap().simplified_display().to_string();
```
Comment (zanieb) [inline crates/uv/src/commands/pip_show.rs:161]: Can we avoid this `unwrap`?

Comment (zanieb) [reply crates/uv/src/commands/pip_show.rs:161]: Do you need to call `to_string`? Isn't display going to be called in `writeln!`?

File: crates/uv/src/commands/pip_show.rs [hunk 2]
```diff
@@ -69,40 +69,77 @@ pub(crate) fn pip_show(
...
+    }
+
+    let distributions = distributions.iter().flatten().collect_vec();
 
     // Like `pip`, if no packages were found, return a failure.
     if distributions.is_empty() {
         return Ok(ExitStatus::Failure);
     }
 
+    // Since Requires and Required-by fields need data parsed from metadata, especially the
```
Comment (zanieb) [inline crates/uv/src/commands/pip_show.rs:99]: @charliermarsh or @konstin could you review this section?

File: crates/uv/tests/pip_show.rs [hunk 3]
```diff
@@ -500,7 +513,75 @@ fn show_editable() -> Result<()> {
     Name: poetry-editable
     Version: 0.1.0
     Location: [WORKSPACE_DIR]/site-packages
+    Editable project location: [EDITABLE_INSTALLS_PREFIX]poetry_editable
     Requires: anyio
+    Required-by:
```
Comment (zanieb) [inline crates/uv/tests/pip_show.rs:518]: I wonder if we should omit these when they're empty

**Comment by zanieb** [comment]
Thanks for contributing!

File: crates/uv/tests/pip_show.rs (hunk 3 above)
Comment (eth3lbert) [reply crates/uv/tests/pip_show.rs:518]: I followed the approach used by pip. I have no preference, either option is fine.

Comment (danielhollas) [reply crates/uv/tests/pip_show.rs:518]: I'd vote to keep pip bwhaviour, otherwise users can wonder if it is even implemented.

Comment (zanieb) [reply crates/uv/tests/pip_show.rs:518]: I don't think being worried that people will wonder if it it's implemented makes sense as a long-term choice. It seems likely if you care about this feature that you'll have at least one package with `Required-by` set. I don't have strong feelings but I'd lean towards omitting empty fields. Curious for others' thoughts though cc @konstin

**Comment by zanieb** [comment]
@danielhollas I wouldn't expand the scope of this pull request, it's better to implement things separately.

File: crates/uv/src/commands/pip_show.rs (hunk 1 above)
Comment (zanieb) [reply crates/uv/src/commands/pip_show.rs:161]: I think so!

**Comment by danielhollas** [comment]
Totally agree! But then please don't close the issue since it's not fully resolved : 😅

**Comment by eth3lbert** [comment]
The implementation itself shouldn't be difficult. However, those fields also come from metadata, which are not stored in Metadata23 (perhaps to keep the footprint size small?). In this case, we need to decide how to proceed: should we simply include more fields here, or introduce a separate struct that provides the full metadata?

**Review by zanieb — APPROVED** [review]
Thanks!

File: crates/uv/src/commands/pip_show.rs (hunk 2 above)
Comment (konstin) [reply crates/uv/src/commands/pip_show.rs:99]: Sorry i missed the ping, code looks good

File: crates/uv/tests/pip_show.rs (hunk 3 above)
Comment (konstin) [reply crates/uv/tests/pip_show.rs:518]: I don't have an opinion on that, we could add `(none)` as placeholder if it helps

**Pull request merged**